pkg/providers/imagefamily/bootstrap/parser_helpers.go

```diff
@@ -563,33 +552,60 @@ func getGpuDriverVersion(vmSize string) string {
 // IsSgxEnabledSKU determines if an VM SKU has SGX driver support.
 func getIsSgxEnabledSKU(vmSize string) bool {
 	switch vmSize {
-	case "Standard_DC2s", "Standard_DC4s":
+	case vmSizeStandardDc2s, vmSizeStandardDc4s:
 		return true
 	}
 	return false
 }
 
 func getShouldConfigureHTTPProxy(httpProxyConfig *nbcontractv1.HTTPProxyConfig) bool {
+	if httpProxyConfig == nil {
```

Comment:
the getters below will return `""` if the config is nil, so we don't need this check here :)